CONTRIBUTING.md

```diff
@@ -0,0 +1,137 @@
+| Service User (\*)             | Service Consumer               | This person can access services by contacting a gateway and providing a valid API key. |
+
+(\*) A real person can be a Service User and a Service Provider at the same time.
+
+
+
+## Contributor License Agreement
+
+### License
+
+By contributing, you agree that your contributions will be licensed under Apache License 2.0. When you submit code changes, your submissions are understood to be under the same Apache License 2.0 that covers the project.
+
+
+
+### Signing Instructions
+
+When submitting a pull request for the first time, you will need to agree to the contributor license agreement (CLA). 
+
+* Individual private persons must agree to the [CLA for individuals](https://github.com/coatrack/coatrack/blob/master/cla/latest/coatrack-individual-cla.pdf) and 
+
+* Organizations must agree to the [CLA for entities](https://github.com/coatrack/coatrack/blob/master/cla/latest/coatrack-entity-cla.pdf). 
+
+If you agree to the respective CLA, you have to create a pull request including a file that officially and transparently documents your intention to sign the CLA:
+
+1. Copy, rename and move [signing-template.md](https://github.com/coatrack/coatrack/blob/master/cla/signing-template.md) to `/cla/contributors/<your-github-username>.md`. 
+2. Fill in the missing information, including the SHA-256 checksum of the CLA PDF file you want to sign, and add this file to your first pull request. The checksum clearly documents the version of the CLA that you agree to.
+3. After you created the pull request, please send an email to `info@coatrack.eu` to inform us about the pull request. 
+
+By merging your pull request, the information that you agreed to the CLA will become part of the official public code base of CoatRack.
+
+
+
+### Structure of the 'cla' folder
+
+For transparency reasons, we will briefly explain here how our CLA was created and how the signing process works. Harmony provides [templates](http://harmonyagreements.org/agreements.html) for CLA's in ODT format which are stored in the [original](https://github.com/coatrack/coatrack/tree/master/cla/source/original) folder. Based on those templates we made custom versions, adapted to our CoatRack requirements, which are stored in the [customized](https://github.com/coatrack/coatrack/tree/master/cla/source/customized) folder. If you want to see the differences between the original and our customized version, you can easily compare the two ODT files (e.g. by using the document comparison feature of LibreOffice). 
+
+The customized CLA ODT's are converted to PDF's and stored in the [latest](https://github.com/coatrack/coatrack/tree/master/cla/latest) folder. These PDF files are the files that have to be signed/hashed in order to be allowed to contribute to CoatRack.
```

Comment:
This is the most important information for most people, so I would put this first, and only afterwards explain  the "transparency information" (as that is only for a sub-group of people).